utility/src/main/java/org/oppia/android/util/logging/firebase/FirebaseEventLogger.kt

```diff
@@ -18,30 +17,29 @@ private const val COUNTRY_USER_PROPERTY = "COUNTRY"
 @Singleton
 class FirebaseEventLogger(
   private val firebaseAnalytics: FirebaseAnalytics,
-  private val eventBundleCreator: EventBundleCreator,
-  context: Context
+  private val eventBundleCreator: EventBundleCreator
 ) : EventLogger {
   private var bundle = Bundle()
-  private val connectivityManager: ConnectivityManager =
-    context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+
+  @Inject
+  lateinit var networkConnectionUtil: NetworkConnectionUtil
 
   /** Logs an event to Firebase Analytics with [NETWORK_USER_PROPERTY] and [COUNTRY_USER_PROPERTY]. */
   override fun logEvent(eventLog: EventLog) {
     bundle = eventBundleCreator.createEventBundle(eventLog)
     firebaseAnalytics.logEvent(eventLog.actionName.toString(), bundle)
     // TODO(#3792): Remove this usage of Locale.
     firebaseAnalytics.setUserProperty(COUNTRY_USER_PROPERTY, Locale.getDefault().displayCountry)
-    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
-      val networkCapabilities = connectivityManager.getNetworkCapabilities(
-        connectivityManager.activeNetwork
-      )
-      firebaseAnalytics.setUserProperty(
-        NETWORK_USER_PROPERTY, networkCapabilities?.transportInfo.toString()
-      )
-    } else {
-      firebaseAnalytics.setUserProperty(
-        NETWORK_USER_PROPERTY, connectivityManager.activeNetworkInfo?.typeName
-      )
+    firebaseAnalytics.setUserProperty(NETWORK_USER_PROPERTY, getNetworkStatus())
+  }
+
+  private fun getNetworkStatus(): String {
+    return when (networkConnectionUtil.getCurrentConnectionStatus()) {
+      NetworkConnectionUtil.ProdConnectionStatus.LOCAL ->
+        NetworkConnectionUtil.ProdConnectionStatus.LOCAL.name
+      NetworkConnectionUtil.ProdConnectionStatus.CELLULAR ->
+        NetworkConnectionUtil.ProdConnectionStatus.CELLULAR.name
+      else -> NetworkConnectionUtil.ProdConnectionStatus.NONE.name
```

Comment:
This seems sensible, though the only thing we should probably do is add string values to the enums that we log, instead, to ensure they're stable (e.g. if we rename the enum constants in the future, we wouldn't want the log values to change).

Please also add tests to verify the exact expected string is logged in the expected circumstances.

Reply:
Updated